### PR TITLE
COMMERCE-12062 Inconsistent behavior with cart total rule in discount portlet

### PR DESCRIPTION
**Pull request by victorcdp**
[COMMERCE-12062](https://liferay.atlassian.net/browse/COMMERCE-12062), [COMMERCE-12091](https://liferay.atlassian.net/browse/COMMERCE-12091)

@andrea-ale-sbarra Can you review this one? Sending with the simpler solution mentioned by Luca in the thread

### CODE REVIEW

**Comment by victorcdp** [comment]
ci:test:sf

**Comment by victorcdp** [comment]
ci:test:relevant

**Comment by victorcdp** [comment]
ci:test:commerce

**Review by alessiorendina — CHANGES_REQUESTED** [review]
Hey @victorcdp please check my comments, thanks.

File: modules/apps/commerce/commerce-discount-service/src/main/java/com/liferay/commerce/discount/service/impl/CommerceDiscountRuleLocalServiceImpl.java [hunk 1]
```diff
@@ -74,6 +75,12 @@ public CommerceDiscountRule addCommerceDiscountRule(
 		commerceDiscountRule.setCommerceDiscountId(commerceDiscountId);
 		commerceDiscountRule.setType(type);
 
+		if (type.equals(CommerceDiscountRuleConstants.TYPE_CART_TOTAL)) {
```
[the remark this answers is not in the document]
Comment (luca-pellizzon) [reply modules/apps/commerce/commerce-discount-service/src/main/java/com/liferay/commerce/discount/service/impl/CommerceDiscountRuleLocalServiceImpl.java:78]: Agree with Alessio.
Also, do we really need a constant to set a 0? It is something olny used once and also the value will never change in the future

**Comment by victorcdp** [comment]
ci:test:relevant

**Comment by victorcdp** [comment]
ci:test:commerce

**Review by alessiorendina — APPROVED** [review]

**Comment by brianikim** [comment]
https://github.com/liferay-commerce/liferay-portal/pull/4129

**Comment by brianikim** [comment]
ci:close

**Pull request closed without merging**